controller/generic/metacontroller.go

```diff
@@ -82,19 +84,15 @@ type ConfigMetaController struct {
 	// controllers
 	Configs []*v1alpha1.GenericController
 
-	// Total timeout for any condition to succeed.
-	//
-	// NOTE:
-	//	This is currently used to load config that is required
-	// to run Metac.
-	WaitTimeoutForCondition time.Duration
+	// This will allow executing start logic to be retried
+	// indefinitely till all the watch controllers are started
+	RetryIndefinitelyForStart *bool
```

Comment:
`RetryIndefinitelyForStart` :arrow_right: I would either leave `RetryIndefinitely` or change suffix `ForStart` to `RetryIndefinitelyUntilSucceed` or something like that